Snakefile

```diff
@@ -62,103 +71,45 @@ def metadata_by_wildcards(wildcards):
-        }
-    return date[w.subtype][w.time]
-
-def traits_params(w):
-    if w.subtype=="h5n1-cattle-outbreak":
-        if w.segment=='genome':
-            return "--columns division --sampling-bias-correction 5"
-        return "--columns region country" # same as GISAID H5N1 builds
-
-    columns = {'h5nx':'region','h5n1': 'region country', 'h7n9': 'country division', 'h9n2': 'region country'}
-    return f"--columns {columns[w.subtype]}"
-
-# The following std-dev is used for all (segment) builds which use a specified clock rate
-CLOCK_STD_DEV = 0.00211
-
-CLOCK_RATES = {
-    "h5nx": {
-        "2y": {
-            'pb2': (0.00287, CLOCK_STD_DEV),
-            'pb1': (0.00267, CLOCK_STD_DEV),
-            'pa': (0.00238, CLOCK_STD_DEV),
-            'ha': (0.0048, CLOCK_STD_DEV),
-            'np': (0.0022, CLOCK_STD_DEV),
-            'na': (0.0028, CLOCK_STD_DEV),
-            'mp': (0.0017, CLOCK_STD_DEV),
-            'ns': (0.0017, CLOCK_STD_DEV),
-        },
-        "all-time": {}
-    },
-    "h5n1": {
-        "2y": {
-            'pb2': (0.00287, CLOCK_STD_DEV),
-            'pb1': (0.00264, CLOCK_STD_DEV),
-            'pa': (0.00248, CLOCK_STD_DEV),
-            'ha': (0.00455, CLOCK_STD_DEV),
-            'np': (0.00252, CLOCK_STD_DEV),
-            'na': (0.00349, CLOCK_STD_DEV),
-            'mp': (0.00191, CLOCK_STD_DEV),
-            'ns': (0.00249, CLOCK_STD_DEV),
-        },
-        "all-time": {}
-    },
-    "h7n9": {
-        "2y": {},
-        "all-time": {}
-    },
-    "h9n2": {
-        "2y": {},
-        "all-time": {}
-    }
-}
-CLOCK_RATES['h5n1-cattle-outbreak'] = {
-    'default': {
-        **CLOCK_RATES['h5n1']['2y'],
-        'genome': calc_genome_clock_rate(
-            {seg: data[0] for seg,data in CLOCK_RATES['h5n1']['2y'].items()},
-            {'pb2': 2341, 'pb1': 2341, 'pa': 2233, 'ha': 1760, 'np': 1565, 'na': 1458, 'mp': 1027, 'ns': 865}
-        ),
-    }
-}
-
-def clock_rate(w):
-    assert w.subtype in CLOCK_RATES, f"Subtype {w.subtype!r} doesn't have any associated clock information"
-
-    assert w.time in CLOCK_RATES[w.subtype], f"Time window {w.time!r} doesn't have any associated clock information for subtype {w.subtype!r}"
-
-    if w.segment not in CLOCK_RATES[w.subtype][w.time]:  # infer a clock rate, so return the empty string
+
+def get_config(rule_name, rule_key, wildcards, segment=None, fallback="ALL"):
+    assert rule_name in config, f"Config missing top-level {rule_name} key"
+    assert rule_key in config[rule_name], f"Config missing entry for {rule_name}.{rule_key}"
+    try:
+        return config[rule_name][rule_key][wildcards.subtype][wildcards.time]
+    except KeyError:
+        assert fallback in config[rule_name][rule_key], f"config.{rule_name!r}.{rule_key!r} either needs " \
+            f"an entry for {wildcards.subtype!r}.{wildcards.time!r} added or a (default) {fallback!r} key."
+        return config[rule_name][rule_key][fallback]

```

Comment:
ok - `FALLBACK` seems fine to me - I didn't want to use "default" because we are already using this word for the time wildcard in NCBI builds and I didn't want any possibility of conflation.

Reply:
Moved to `FALLBACK` in latest push